core/libraries/batch/JobHandlers/RegistrationsReport.php

```diff
@@ -57,43 +57,59 @@ class RegistrationsReport extends JobHandlerFile
+            throw new BatchRequestException(
+                esc_html__('You do not have permission to view registrations', 'event_espresso')
+            );
         }
         $filepath = $this->create_file_from_job_with_name(
             $job_parameters->job_id(),
             $this->get_filename()
         );
         $job_parameters->add_extra_data('filepath', $filepath);
-        $query_params = apply_filters('FHEE__EE_Export__report_registration_for_event', array(
-            array(
-                'OR'                 => array(
-                    // don't include registrations from failed or abandoned transactions...
-                    'Transaction.STS_ID' => array(
-                        'NOT IN',
-                        array(
-                            EEM_Transaction::failed_status_code,
-                            EEM_Transaction::abandoned_status_code,
-                        ),
-                    ),
-                    // unless the registration is approved, in which case include it regardless of transaction status
-                    'STS_ID'             => EEM_Registration::status_id_approved,
-                ),
-                'Ticket.TKT_deleted' => array('IN', array(true, false)),
-            ),
-            'order_by'   => array('Transaction.TXN_ID' => 'asc', 'REG_count' => 'asc'),
-            'force_join' => array('Transaction', 'Ticket', 'Attendee'),
-            'caps'       => EEM_Base::caps_read_admin,
-        ), $event_id);
+
+
+        $filter_params = maybe_unserialize($job_parameters->request_datum('filters', []));
+
+        $query_params = is_array($filter_params) && ! empty($filter_params)
+            ? $filter_params
+            : [
+                [
+                    'OR'                 => [
+                        // don't include registrations from failed or abandoned transactions...
+                        'Transaction.STS_ID' => [
+                            'NOT IN',
+                            [
+                                EEM_Transaction::failed_status_code,
+                                EEM_Transaction::abandoned_status_code,
+                            ],
+                        ],
+                        // unless the registration is approved, in which case include it regardless of transaction status
+                        'STS_ID'             => EEM_Registration::status_id_approved,
+                    ],
+                    'Ticket.TKT_deleted' => ['IN', [true, false]],
+                ],
+                'order_by'   => ['Transaction.TXN_ID' => 'asc', 'REG_count' => 'asc'],
+                'force_join' => ['Transaction', 'Ticket', 'Attendee'],
+                'caps'       => EEM_Base::caps_read_admin,
+            ];
+
         if ($event_id) {
             $query_params[0]['EVT_ID'] = $event_id;
         } else {
             $query_params['force_join'][] = 'Event';
```

Comment:
Previously, this was only set on the 'default' query params, now it's set on both the default and filtered query params.

This means line 104 no longer jumps in and sets the 'correct' force_join for the filtered registrations, this is why the attendee fields no longer show. The query no longer returns attendee data from the DB.